client/asset/btc/rpcclient.go

```diff
@@ -495,10 +495,31 @@ func (wc *rpcClient) listUnspent() ([]*ListUnspentResult, error) {
 	return unspents, wc.call(methodListUnspent, anylist{uint8(0)}, &unspents)
 }
 
+// lockAllUnspent unlocks all locked outputs for spending again. If unlock is
+// true btc will return 'true' if successful. If unlock is false btc will also
+// return 'true' but do nothing so a warning is issued below.
+func (wc *rpcClient) lockAllUnspent(unlock bool) error {
+	if !unlock {
+		wc.log.Warn("attempting to lock all unspent outputs 'lockunspent false'")
+	}
+	var rpcReturn bool
+	err := wc.call(methodLockUnspent, anylist{unlock}, &rpcReturn)
+	if err != nil {
+		return fmt.Errorf("'lockunspent %v': %v: ", unlock, err)
+	}
+	if !rpcReturn {
+		return fmt.Errorf("'lockunspent %v' unsuccessful: RPC returns %v", unlock, rpcReturn)
+	}
+	return nil
+}
+
 // lockUnspent locks and unlocks outputs for spending. An output that is part of
 // an order, but not yet spent, should be locked until spent or until the order
 // is canceled or fails.
 func (wc *rpcClient) lockUnspent(unlock bool, ops []*output) error {
+	if ops == nil {
+		return wc.lockAllUnspent(unlock)
+	}
 	var rpcops []*RPCOutpoint // To clear all, this must be nil->null, not empty slice.
 	for _, op := range ops {
```

Comment:
@chappjc, please correct me if I'm wrong, but since we are now performing a nil check earlier, I believe this comment is no longer applicable.

Reply:
I agree but thought it not my place to delete it .. although now I could if you like

Reply:
> I agree but thought it not my place to delete it .. although now I could if you like

I think the `ops` parameter should own this comment instead of `rpcops`.

Reply:
Then since there is no comment at function level, and the comment is never now going to be true would you agree that I should remove it. I removed on my local and all looks sensible to me.

Reply:
Would referencing this doc (from the `btc_test.go` test case) in `rpcClient.lockUnspent`or the function level doc of `rpcClient.lockUnspent` be useful? Keeping the main path clutter free

![image](https://github.com/decred/dcrdex/assets/126670673/58c5d572-9f80-4f4e-bf63-36f6810ac785)

Reply:
Fixed in https://github.com/decred/dcrdex/pull/2397/commits/b7354fc99d3ada4141e9176697ec6a0b928b0f4d
Regularized comments as per approval